prov/rxd/src/rxd_av.c

```diff
@@ -103,68 +86,39 @@ static int rxd_av_set_addrlen(struct rxd_av *av, const void *addr)
 	return ret;
 }
 
-static fi_addr_t rxd_av_dg_addr(struct rxd_av *av, fi_addr_t fi_addr)
+int rxd_av_insert_fi_addr(struct rxd_av *av, fi_addr_t *dg_addr, 
```

Comment:
I would move this whole function back into the rxd_av_insert function instead of separating it out since you're calling fi_av_remove on the dg_addr which wasn't inserted in this function and then it would just be a one list ofi_av_insert_addr. I think you can handle the error cases better in the other function.